var/spack/repos/builtin/packages/rust/package.py

```diff
@@ -91,9 +94,33 @@ def setup_build_environment(self, env):
         ar = which("ar", required=True)
         env.set("AR", ar.path)
 
-        # Manually inject the path of openssl's certs for build.
-        certs = join_path(self.spec["openssl"].prefix, "etc/openssl/cert.pem")
-        env.set("CARGO_HTTP_CAINFO", certs)
+        # Manually inject the path of openssl's certs for build
+        # if certs are present on system via Spack or via external
+        # openssl.
+        def get_test_path(p):
+            certs = join_path(p, "cert.pem")
+            if os.path.exists(certs):
+                return certs
+            return None
+
+        # find certs, don't set if no file is found in case
+        # ca-certificates isn't installed
+        certs = None
+        openssl = self.spec["openssl"]
+        if openssl.external:
+            try:
+                output = which("openssl", required=True)("version", "-d", output=str, error=str)
+                openssl_dir = re.match('OPENSSLDIR: "([^"]+)"', output)
+                if openssl_dir:
+                    certs = get_test_path(openssl_dir.group(1))
+            except ProcessError:
+                pass
```

Comment:
Did we consider moving this into the `openssl` package (seems more related to a behavior `openssl` has when external than to Rust)?

Reply:
@trws and I talked about it, but this is really a one off fix for `rust`/`rust-bootstrap` who ship their own statically linked openssl as part of the bootstrap binary. That embedded openssl needs access to custom site firewall certs if they exist on the host system so that it doesn't realize the site firewall is performing a man-in-the-middle of the cargo index.

Reply:
If we were able to use either the system/spack installed openssl (like all other packages) this wouldn't be a problem.

@trws did we consider putting this in `rust-bootstrap` instead of `rust` and having it set up for dependent package environments? (Rust)

Reply:
Also note there was already logic for this here, this is just my attempt to refine it to cover the external case a bit more reliably.

Reply:
Fair enough, thanks.